platform/android/java/lib/src/org/godotengine/godot/utils/InputUtils.java

```diff
@@ -0,0 +1,42 @@
+/*************************************************************************/
+/*  InputUtils.java                                                      */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+package org.godotengine.godot.utils;
+import android.view.MotionEvent;
+
+public final class InputUtils {
+	private InputUtils() {}
+	/**
+	 * Makes BUTTON_BACK and BUTTON_FORWARD mask to be equal to godot's BUTTON_MASK_XBUTTON
+	 */
+	public static int fixButtonsStateMask(int buttonMask) {
+		return (buttonMask & ~(MotionEvent.BUTTON_BACK | MotionEvent.BUTTON_FORWARD)) | (((MotionEvent.BUTTON_BACK | MotionEvent.BUTTON_FORWARD) & buttonMask) << 4);
```

Comment:
What is `<< 4` for? It seems risky to have an undocumented constant here, especially if it refers to a constant from the c++ side.

Reply:
The android's bitmasks are constant, and godot's should also be, as commented on the code shifting the difference bitmask from android mask should be safe, but I don't think putting this in the cpp code as it makes sense to fix it here where I put the afterward the wheel mask... It can be in the cpp side but it'll make the code more complicated (more specifically because of the scrolling side)